docs/modules/trino/pages/usage-guide/catalogs/delta-lake.adoc

```diff
@@ -0,0 +1,34 @@
+= Delta Lake
+
+[Delta Lake](https://delta.io/) is an open-source storage framework that enables building a Lakehouse architecture with support for multiple compute engines.
+It depends on a Hive metastore being present and makes use of the same metastore configMap used by the Hive connector.
+
+== Example Delta Lake catalog configuration
+
+[source,yaml]
+----
+apiVersion: trino.stackable.tech/v1alpha1
+kind: TrinoCatalog
+metadata:
+  # The name of the catalog as it will appear in Trino
+  name: delta
+  labels:
+    trino: simple-trino
+spec:
+  connector:
+    # Specify deltaLake here when defining a Delta Lake catalog
+    deltaLake:
+      metastore:
+        configMap: simple-hive-derby
```

Comment:
Would not use derby anymore in examples.
```suggestion
        configMap: simple-hive
```

Reply:
We actually have the following. WDYT, should we change it or leave it as is for now?
```
➜  trino-operator git:(feat/delta-lake) ✗ ack 'hive-derby'
docs/modules/trino/examples/usage-guide/trino-insecure.yaml
12:        configMap: simple-hive-derby
37:  name: simple-hive-derby

docs/modules/trino/examples/usage-guide/trino-secure-internal-tls.yaml
12:        configMap: simple-hive-derby
72:  name: simple-hive-derby

docs/modules/trino/examples/usage-guide/trino-secure-tls.yaml
12:        configMap: simple-hive-derby
72:  name: simple-hive-derby

docs/modules/trino/examples/usage-guide/trino-secure-tls-only.yaml
12:        configMap: simple-hive-derby
52:  name: simple-hive-derby

docs/modules/trino/pages/usage-guide/catalogs/iceberg.adoc
21:        configMap: simple-hive-derby

docs/modules/trino/pages/usage-guide/catalogs/index.adoc
26:        configMap: simple-hive-derby
48:        configMap: simple-hive-derby

examples/simple-trino-cluster.yaml
32:        configMap: simple-hive-derby
37:  name: simple-hive-derby

examples/simple-trino-cluster-hive-ha-s3.yaml
36:        configMap: simple-hive-derby
43:  name: simple-hive-derby

examples/simple-trino-cluster-authentication-opa-authorization-s3.yaml
122:        configMap: simple-hive-derby
127:  name: simple-hive-derby

examples/simple-trino-cluster-s3.yaml
34:        configMap: simple-hive-derby
41:  name: simple-hive-derby
```

Reply:
Since this is not for production i would only have it in the actual derby example. Not sure but i think we wanted to remove derby at some point anyways?

Reply:
I removed the derby suffix from all docs that don't actually contain a HMS using derby in 934240613a5869cabe108a0555d47be0bcc6c506